components/form/builder/src/reducer/rules.js

```diff
@@ -118,60 +118,74 @@ export const shouldApplyRule = (fields, changeField, locale) => when => {
   return isValid
 }
 
-export const fetchRemoteFields =
-  (
-    fields,
-    formID,
-    baseAPIURL,
-    responseInterceptor,
-    requestInterceptor,
-    extraParams
-  ) =>
-  async fieldsToChanges => {
-    const remoteFieldsToChange = await Promise.all(
-      Object.entries(fieldsToChanges)
-        .filter(([field, nextValue]) => nextValue.remote === REMOTE)
-        .map(async ([field, nextValue]) => {
-          const defaultUrl = encodeURI(
-            `${baseAPIURL}/fieldrules/${field}?${fieldsToQP(fields, formID)}`
-          )
-
-          const defaultConfig = {url: defaultUrl}
-
-          const requestInterceptorConfig = await requestInterceptor({
-            baseAPIURL,
-            fieldID: field,
-            fields,
-            extraParams
+export const fetchRemoteFields = (
+  fields,
+  formID,
+  baseAPIURL,
+  responseInterceptor,
+  requestInterceptor,
+  extraParams
+) => async fieldsToChanges => {
+  const remoteFieldsToChange = await Promise.all(
+    Object.entries(fieldsToChanges)
+      .filter(([field, nextValue]) => nextValue.remote === REMOTE)
+      .map(async ([field, nextValue]) => {
+        const defaultUrl = encodeURI(
+          `${baseAPIURL}/fieldrules/${field}?${fieldsToQP(fields, formID)}`
+        )
+
+        const defaultConfig = {url: defaultUrl}
+
+        const {
+          callback,
+          ...requestInterceptorConfig
+        } = await requestInterceptor({
+          baseAPIURL,
+          fieldID: field,
+          fields,
+          extraParams
+        })
+
+        // Config must be follow the Axios pattern
+        // https://github.com/axios/axios
+        const config = requestInterceptorConfig || defaultConfig
+
+        const {remote, ...resetValue} = nextValue
+
+        if (callback) {
+          const response = await callback()
+
+          const nextJSON = await responseInterceptor({
+            field,
+            url: '',
```

Comment:
I leave the empty string in case it's expected to always be the `url` param on the `responseInterceptor` call, in order to not make any breaking change. 

In MA, for example, we check the incoming URL if includes some path to perform some actions.